test/fancy_select/index.js

```diff
@@ -0,0 +1,118 @@
+const { createElement } = require('react');
+const { assert } = require('chai');
+const { mount, shallow } = require('enzyme');
+
+const FancySelect = require('../../lib/fancy_select').default;
+
+
+describe('FancySelect', () => {
+  it('renders FancySelect', () => {
+    const props = {
+      defaultValue: 1,
+      options: [
+        { imageClass: 'c-sandbox-fancy_image_1', key: 'Eins', value: 1 },
+        { imageClass: 'c-sandbox-fancy_image_2', key: 'Zwei', value: 2 },
+      ],
+    };
+    const wrapper = mount(createElement(FancySelect, props));
+    const instance = wrapper.instance();
+
+    assert.equal(wrapper.find('ul .c-fancy_select-list').length, 1, 'FancySelect was rendered');
+    assert.equal(wrapper.find('li .c-fancy_select-item').length, 2, 'two options were rendered');
+    assert.equal(instance.state.value, 1, 'value is set');
+    assert.equal(instance.state.index, 0, 'index is set');
+
+    wrapper.find('li .c-fancy_select-item').last().simulate('click');
+
+    assert.equal(instance.state.value, 2, 'value was changed');
+    assert.equal(instance.state.index, 1, 'index was changed');
+    assert.isTrue(
+      wrapper.find('li .c-fancy_select-item').last().hasClass('is-active'),
+      'active option has active class',
+    );
+
+    wrapper.unmount();
+  });
+
+  it('defaultValue', () => {
+    const props = {
+      defaultValue: 1,
+      options: [
+        { imageClass: 'c-sandbox-fancy_image_1', key: 'Eins', value: 555 },
+        { imageClass: 'c-sandbox-fancy_image_2', key: 'Zwei', value: 2 },
+      ],
+    };
+    const wrapper = shallow(createElement(FancySelect, props));
+    const instance = wrapper.instance();
+
+    assert.equal(instance.state.value, 555, 'default value is correct');
+    assert.equal(instance.state.index, 0, 'index is correct');
+  });
+
+  it('setValue', () => {
+    const props = {
+      defaultValue: 1,
+      options: [
+        { imageClass: 'c-sandbox-fancy_image_1', key: 'Eins', value: 1 },
+        { imageClass: 'c-sandbox-fancy_image_2', key: 'Zwei', value: 222 },
+      ],
+    };
+    const wrapper = shallow(createElement(FancySelect, props));
+    const instance = wrapper.instance();
+
+    assert.equal(instance.getValue(), 1, 'default value is set');
+    assert.equal(instance.state.index, 0, 'index is correct');
```

Comment:
why do u check defaultValue here? u checked it in case above

Reply:
`+`